### PR TITLE
Implement permissions decorator for cogs.

### PR DESCRIPTION
**Pull request by benwoo1110**
## About this pull request

Implement permissions decorator for cogs and updated docs on it to be clearer. Not sure why I didn't add it previously... 

To test out, `pip install git+https://github.com/benwoo1110/discord-py-interactions@cog-permissions-decorator`

## Checklist

- [x] I've run the `pre_push.py` script to format and lint code.
- [x] I've checked this pull request runs on `Python 3.6.X`.
- [ ] This fixes something in [Issues](https://github.com/eunwoo1104/discord-py-slash-command/issues).
    - Issue:
- [c] This adds something new.
- [ ] There is/are breaking change(s).
- [x] (If required) Relevant documentation has been updated/added.
- [ ] This is not a code change. (README, docs, etc.)

### CODE REVIEW

**Review by LordOfPolls — APPROVED** [review]

**Comment by LordOfPolls** [comment]
Works fine, LGTM. 

Waiting on one more review

**Comment by FayeDel** [comment]
Tested, works on my end, LGTM.

**Pull request merged**
